usb: cdc_acm: compilation error without UART

**Describe the bug**

The USB CDC ACM driver cannot be used on a board without a physical UART peripheral being enabled.
Attempting to do so results in the following error:
```
-- Configuring done
CMake Error at ../../../../cmake/extensions.cmake:416 (add_library):
  No SOURCES given to target: drivers__serial
Call Stack (most recent call first):
  ../../../../cmake/extensions.cmake:393 (zephyr_library_named)
  ../../../../drivers/serial/CMakeLists.txt:3 (zephyr_library)
```

`CONFIG_USB_CDC_ACM` selects `SERIAL_HAS_DRIVER`, which adds the `serial/CMakeLists.txt` file.
Without a UART peripheral enabled however, none of the `zephyr_library_sources_ifdef`'s result in a source being added.
This leads to the above error.

**To Reproduce**
Steps to reproduce the behavior:
1. add the following overlay as `zephyr/samples/subsys/usb/console/nrf52840dk_nrf52840.overlay`
```
&uart0 {
	status = "disabled";
};

&uart1 {
	status = "disabled";
};
```
2. west build -b nrf52840dk_nrf52840 zephyr/samples/subsys/usb/console
3. See error

**Expected behavior**
Compilation should not fail when a UART peripheral is not being compiled in.

**Impact**
UART peripherals need to be enabled despite not being used, increased RAM and ROM footprint.

**Environment (please complete the following information):**
 - Zephyr v2.4.99

--- a/drivers/serial/CMakeLists.txt
+++ b/drivers/serial/CMakeLists.txt
@@ -36,6 +36,7 @@ zephyr_library_sources_ifdef(CONFIG_UART_XLNX_UARTLITE uart_xlnx_uartlite.c)
 zephyr_library_sources_ifdef(CONFIG_UART_XMC4XXX uart_xmc4xxx.c)
 zephyr_library_sources_ifdef(CONFIG_UART_NPCX uart_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_UART_APBUART uart_apbuart.c)
+zephyr_library_sources_ifdef(CONFIG_USB_CDC_ACM ${ZEPHYR_BASE}/misc/empty_file.c)
 
 zephyr_library_sources_ifdef(CONFIG_USERSPACE   uart_handlers.c)
 

